packages/wonder-blocks-core/src/hooks/use-isomorphic-layout-effect.ts

```diff
@@ -0,0 +1,34 @@
+import * as React from "react";
+
+import Server from "../util/server";
+
+/**
+ * Hook that works like useLayoutEffect but does not error on the server.
+ *
+ * Effects from `useLayoutEffect` cause errors on the server. Rather than
+ * erroring, we want the server side to just silently no-op like `useEffect`
+ * calls do. This hook allows that.
+ */
+export const useIsomorphicLayoutEffect: typeof React.useLayoutEffect = (
```

Comment:
**note:** This is based on the hook of the same name in webapp except:

1. It does not use `React.useEffect` when server-side; it just invokes a no-op.
2. It does not decide which call to use on import, but instead on first invocation - this is to simplify testing, since things that vary on import require module isolation and what have you when trying to test CSR vs SSR scenarios. It uses a ref so that the decision is only made once, on initial invocation.